elastic/beats
# [Metricbeat/Kibana/stats] Enforce `exclude_usage=true`

## What does this PR do?


Changes to the Metricbeat's `kibana` module: it enforces `exclude_usage=true` (only for the versions it's accepted).

The reason for this change is to disable the usage collection via the `/api/stats` endpoint, as it is no longer needed for their initial Telemetry purposes.

## Why is it important?


We are working towards decoupling Monitoring and Telemetry in Kibana. At the moment, Monitoring holds in the `.monitoring-kibana-*` indices the Kibana's Usage details only to be able to report them when sending the telemetry reports. This data is already sent to the remote telemetry cluster via Kibana's `local` collectors, so there's no need for this duplication in the efforts. Thanks to these changes, Monitoring will only maintain in its indices the data they really need for its own features.


## Checklist




## Author's Checklist


- [ ] Not a Go developer myself. Did I mess anything up? 😇 

## How to test this PR locally


Setup Metricbeat with the `kibana-xpack` module enabled and notice it never asks for the `/api/stats?extended=true&legacy=true&exclude_usage=false` request (the last query parameter is always `true`).

## Related issues


- Closes #22651
- Superseds #22654
- Relates https://github.com/elastic/kibana/issues/83521
- Relates https://github.com/elastic/kibana/issues/83626

## Use cases


Explained in the issue #22651

## Screenshots



## Logs
